torch/csrc/jit/passes/xnnpack_rewrite.cpp

```diff
@@ -0,0 +1,121 @@
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/xnnpack_rewrite.h>
+#include <torch/csrc/jit/passes/fuse_linear.h>
+#include <torch/csrc/jit/passes/graph_rewrite_helper.h>
+#include <torch/csrc/jit/passes/subgraph_rewrite.h>
+#include <torch/csrc/jit/ir/subgraph_matcher.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+std::string getFuncName(Value* func_value) {
+  auto func_node = func_value->node();
+  auto func = func_node->output()->type()->expect<FunctionType>()->function();
+  const auto& qname = func->qualname();
+  const auto& name = qname.qualifiedName();
+  auto rdot_idx = name.rfind('.');
+  if (rdot_idx != std::string::npos) {
+    return name.substr(rdot_idx + 1, name.length());
+  } else {
+    return name;
+  }
+}
+
+void insertXNNPACKLinearOp(std::shared_ptr<Graph>& graph) {
+  std::string linear_before_inline = R"(
+    graph(%linear, %input, %weight, %bias):
+        %r = prim::CallFunction(%linear, %input, %weight, %bias)
+        return (%r))";
+  std::string xnnpack_pattern_before_inline = R"(
+    graph(%linear, %input, %weight, %bias):
+        %packed_weight_bias = xnnpack::linear_prepack(%weight, %bias)
+        %res = xnnpack::linear_packed(%input, %packed_weight_bias)
```

Comment:
why is this called `xnnpack::linear_packed`? do you have another function called `xnnpack::linear`?

Reply:
This is how it was called in the original diff that landed xnnpack (https://github.com/pytorch/pytorch/pull/32509), so kept the naming convention.

Reply:
It's to emphasize that the arguments are not normal weights and biases.